app/lib/actions/candlepin/candlepin_listening_service.rb

```diff
@@ -40,7 +40,9 @@ def close
       end
 
       def retrieve
-        return @receiver.fetch(TIMEOUT)
+        result = @receiver.fetch(TIMEOUT)
+        @session.acknowledge(:message => result, :sync => true)
```

Comment:
The window for an issue here is very small. I dont know if its worth putting this in a try/except and calling '@session.release' on the message in the except block, I could go either way on it.

```
      # Releases the message, which allows the broker to attempt to
      # redeliver it.
      #
      # NOTE: A message connot be released once it has been acknowled.
```